### PR TITLE
ROX-9754: Required ingress network policy default policy

### PR DESCRIPTION
**Pull request by fredrb**
## Description

This PR adds the default policy for Network Policy criteria. It also introduces some mechanism to filter default system policies loaded based on feature flags enabled.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~~Evaluated and added CHANGELOG entry if required~~
- [ ] ~~Determined and documented upgrade steps~~
- [ ] ~~Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

## Testing Performed

- New unit test for feature flag mechanism
- Deployed two clusters (with and without feature flag) and checked that default policy is present respectively

## Policy Display:

On the policy list:
image

On the policy detail screen:
image

**(!)** Please note that policy criteria is empty because we still don't have the UI descriptors for it in place. These are being done in parallel in #1194.

### CODE REVIEW

File: pkg/defaults/policies/files/deployment_has_ingress_network_policy.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "criteriaLocked": true,
+  "mitreVectorsLocked": true,
+  "isDefault": true
+}
```
Comment (vikin91) [inline pkg/defaults/policies/files/deployment_has_ingress_network_policy.json:147]: Missing newline

**Review by vikin91 — APPROVED** [review]
LGTM! This is only ingress - what about egress?

File: pkg/defaults/policies/files/deployment_has_ingress_network_policy.json (hunk 1 above)
Comment (fredrb) [reply pkg/defaults/policies/files/deployment_has_ingress_network_policy.json]: Done

**Comment by fredrb** [comment]
> LGTM! This is only ingress - what about egress?

@vikin91 the agreement with product folks was that the default policy provided at this point will cover `ingress` only. We already provide the `egress` field, meaning that if users are interested in that as well they'll need to either create a new policy or clone this one and add the additional field.

**Pull request merged**